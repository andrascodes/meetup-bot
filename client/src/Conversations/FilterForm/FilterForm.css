.SearchLabel, .DatePickerLabel, .SelectLabel, .DateInput, .SearchInput {
  margin-right: 5px;
}

/*.SearchField, .DateRangePicker {
  display: inline;
}*/

/* Medium Devices, Desktops */
/*@media only screen and (max-width : 992px) {
  .SearchField, .DateRangePicker {
    display: block;
  }
}*/