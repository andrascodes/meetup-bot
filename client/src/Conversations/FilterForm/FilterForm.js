import React from 'react'

import './FilterForm.css'

import { SearchField as ParticipantSearch } from './SearchField'
import { DateRangePicker } from './DateRangePicker'
import { SelectField as LabelSelector, SelectField as ErrorSelector } from './SelectField'

export const FilterForm = props => (
  <div className="FilterForm">
    <div className="container-fluid">
      <div className="row">
        <div className="col-sm-6 col-md-6 col-lg-6 text-center">
          <ParticipantSearch 
            options={props.participantList}
            handleInput={props.handleParticipantInput}
            defaultParticipantValue={props.defaultParticipantValue}
          />
        </div>
        <div className="col-sm-6 col-md-6 col-lg-6 text-center">
          <DateRangePicker 
            minStartDate={props.minStartDate}
            maxEndDate={props.maxEndDate}
            defaultStartValue={props.defaultStartValue}
            defaultEndValue={props.defaultEndValue}
            handleStartInput={props.handleStartInput}
            handleEndInput={props.handleEndInput}
          />
        </div>
      </div>
      <div className="row">
        <div className="col-sm-6 col-md-6 col-lg-6 text-center">
          <LabelSelector 
            name="clusterLabelSelector"
            label="Label:"
            options={props.labelOptions}
            handleChange={props.handleLabelChange}
            defaultValue={props.defaultLabelValue}
          />
        </div>
        <div className="col-sm-6 col-md-6 col-lg-6 text-center">
          <ErrorSelector 
            name="errorSelector"
            label="Error:"
            options={props.errorOptions}
            handleChange={props.handleErrorChange}
            defaultValue={props.defaultErrorValue}
          />
        </div>  
      </div>
      
    </div>

  </div>
)

FilterForm.PropTypes = {
  participantList: React.PropTypes.array.isRequired,
  handleParticipantInput: React.PropTypes.func.isRequired,
  defaultParticipantValue: React.PropTypes.string,

  minStartDate: React.PropTypes.string,
  maxEndDate: React.PropTypes.string,
  defaultStartValue: React.PropTypes.string,
  defaultEndValue: React.PropTypes.string,
  handleStartInput: React.PropTypes.func.isRequired,
  handleEndInput: React.PropTypes.func.isRequired,

  labelOptions: React.PropTypes.array.isRequired,
  handleLabelChange: React.PropTypes.func.isRequired,
  defaultLabelValue: React.PropTypes.string,

  errorOptions: React.PropTypes.array.isRequired,
  handleErrorChange: React.PropTypes.func.isRequired,
  defaultErrorValue: React.PropTypes.string,
}