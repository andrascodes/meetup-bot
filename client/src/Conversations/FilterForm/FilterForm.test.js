import React from 'react'
import renderer from 'react-test-renderer'

import { FilterForm } from './FilterForm'

describe('FilterForm', () => {

  test('renders correctly', () => {
    
    const tree = renderer.create(
      <FilterForm 
        participantList={['123', '456']}
        handleParticipantInput={() => {}}
        defaultParticipantValue={'123'}

        minStartDate={'2017-04-13'}
        maxEndDate={'2017-05-14'}
        defaultStartValue={'2017-04-15'}
        defaultEndValue={'2017-04-20'}
        handleStartInput={() => {}}
        handleEndInput={() => {}}

        labelOptions={[{
          value: 'label1 label2',
          text: 'label1 label2'
        }]}
        handleLabelChange={() => {}}
        defaultLabelValue={'null'}

        errorOptions={[{
          value: 'OAuthException',
          text: 'OAuthException'
        }]}
        handleErrorChange={() => {}}
        defaultErrorValue={'----'}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  test('throws PropType error', () => {
    const render = () => renderer.create(
      <FilterForm
        handleRowClick={(id) => () => {}}
      />
    )
    expect(render).toThrowError(`The prop 'options' has to be defined, pass an empty array ([])`)
  })

})