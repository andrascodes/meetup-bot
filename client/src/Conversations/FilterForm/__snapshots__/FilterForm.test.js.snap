exports[`FilterForm renders correctly 1`] = `
<div
  className="FilterForm">
  <div
    className="container-fluid">
    <div
      className="row">
      <div
        className="col-sm-6 col-md-6 col-lg-6 text-center">
        <div
          className="form-group SearchField">
          <label
            className="SearchLabel"
            htmlFor="participants">
            Participant:
          </label>
          <input
            className="SearchInput"
            defaultValue="123"
            list="participants"
            onInput={[Function]} />
          <datalist
            id="participants">
            <option
              value="123" />
            <option
              value="456" />
          </datalist>
        </div>
      </div>
      <div
        className="col-sm-6 col-md-6 col-lg-6 text-center">
        <div
          className="DateRangePicker">
          <label
            className="DatePickerLabel"
            htmlFor="from">
            Date between:
          </label>
          <input
            className="DateInput"
            defaultValue="2017-04-15"
            id="from"
            max="2017-04-20"
            min="2017-04-13"
            onInput={[Function]}
            type="date" />
          <label
            className="DatePickerLabel"
            htmlFor="to">
            and
          </label>
          <input
            className="DateInput"
            defaultValue="2017-04-20"
            id="to"
            max="2017-05-14"
            min="2017-04-15"
            onInput={[Function]}
            type="date" />
        </div>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-sm-6 col-md-6 col-lg-6 text-center">
        <div
          className="SelectField">
          <label
            className="SelectLabel"
            htmlFor="clusterLabelSelector">
            Label:
          </label>
          <select
            defaultValue="null"
            id="clusterLabelSelector"
            name="clusterLabelSelector"
            onChange={[Function]}>
            <option
              value="label1 label2">
              label1 label2
            </option>
          </select>
        </div>
      </div>
      <div
        className="col-sm-6 col-md-6 col-lg-6 text-center">
        <div
          className="SelectField">
          <label
            className="SelectLabel"
            htmlFor="errorSelector">
            Error:
          </label>
          <select
            defaultValue="----"
            id="errorSelector"
            name="errorSelector"
            onChange={[Function]}>
            <option
              value="OAuthException">
              OAuthException
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;
